osu.Game.Tests/Visual/Gameplay/TestSceneReplayDownloadButton.cs

```diff
@@ -8,34 +8,97 @@
 using osu.Game.Scoring;
 using osu.Game.Users;
 using osu.Framework.Allocation;
+using osu.Framework.Bindables;
 using osu.Game.Rulesets;
 using osu.Game.Screens.Ranking;
+using osuTK.Input;
 
 namespace osu.Game.Tests.Visual.Gameplay
 {
     [TestFixture]
-    public class TestSceneReplayDownloadButton : OsuTestScene
+    public class TestSceneReplayDownloadButton : OsuManualInputManagerTestScene
     {
         [Resolved]
         private RulesetStore rulesets { get; set; }
 
         private TestReplayDownloadButton downloadButton;
 
-        public TestSceneReplayDownloadButton()
+        [Test]
+        public void TestDisplayStates()
         {
-            createButton(true);
+            AddStep(@"create button with replay", () =>
+            {
+                Child = downloadButton = new TestReplayDownloadButton(getScoreInfo(true))
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                };
+            });
+
+            AddUntilStep("wait for load", () => downloadButton.IsLoaded);
+
+            AddStep("click button", () => downloadButton.TriggerClick());
```

Comment:
For what it's worth this step doesn't actually do the same thing as clicking via `ManualInputManager`. This is because `TriggerClick()` sends UI events directly to the `ReplayDownloadButton`, while it is in fact its child - a nested `DownloadButton` - which actually handles click events and runs all of the logic.

Seeing that the other test does this properly though I'm fine with either using `ManualInputManager` or removing.

Reply:
I thought I removed this, oops.